### PR TITLE
Le unified post table, "optimize all queries"

### PR DESCRIPTION
**Pull request by Apogate**
Completed and ready for merging pending the destruction of my dreams.
### What it does:

 "optimizes" all SQL queries by adding `limits` where necessary to cut down on execution time/memory usage/whatever. I am not as experienced with SQL as I am, say, with ur mum, so I don't know how much of an impact this will have performance wise. Probably significant on higher loads. maybe this is a huge flaming wreck and I don't even realize it.

### CODE REVIEW

**Comment by RePod** [comment]
The future is garbage, I want the old future back.

File: _core/admin/delete.php [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+			uses the checkbox deletion method while browsing.
+			
+			targeted actually processes the deletion, deleting the targeted post(s), 
+			their children, reports and destroys any necessary files (HTML and media).
+		
+			It is in desperate need of a rewrite. (meme header)
+		=========================================================================
+*/
+
+
```
Comment (RePod) [inline _core/admin/delete.php:18]: This header...

File: _core/regist/regist.php [hunk 2]
```diff
@@ -117,9 +120,9 @@ private function insert($info) {
         $keys = implode(",",$keys);
         $vals = implode(",",$vals);
 
-        $query = "insert into ".SQLLOG." ($keys) values ($vals)";
+        $query = "INSERT INTO ".SQLLOG." ($keys) VALUES ($vals) ON DUPLICATE KEY UPDATE no = no+1";
         $mysql->query($query);
-        $final = (int) $mysql->result('select last_insert_id()');
+        $final = (UNIFIED_TABLE) ? $mysql->result("SELECT MAX(no) FROM " . SQLLOG . " WHERE board='" . BOARD_DIR . "'") : (int) $mysql->result('SELECT last_insert_id()');
```
Comment (RePod) [inline _core/regist/regist.php]: ~~Does this actually matter? I'm sure `last_insert_id()` also catches `UPDATE`.~~
Apparently not, still really hacky. However the original `INSERT` query should still fire properly?

File: config.php [hunk 3]
```diff
@@ -14,20 +14,28 @@
     The database and tables are created automatically using these values.
     Scroll below to the MySQL Advanced section for additional options.
 */
-define(SQLUSER, 'username');
+
+define(SQLUSER, 'root');
```
Comment (RePod) [inline config.php]: Triggered.

File: config.php [hunk 4]
```diff
@@ -14,20 +14,28 @@
...
+
+define(UNIFIED_TABLE, false);	
 
 /*
     Something descriptive.
 */
 
 define(PANEL_PASS, 'CHANGEME');  //Staff action key  (CHANGE THIS YO)
-define(SITE_ROOT, 'example.com');//Site domain.
+define(SITE_ROOT, 'example.org');//Site domain.
```
Comment (RePod) [inline config.php]: `.net`

File: test.php [hunk 5]
```diff
@@ -257,18 +257,21 @@
                     echo "<strong>$db</strong> database already exists.<br>";
                 }
 
+				
+				$noType = (UNIFIED_TABLE) ? "no int" : "no int not null auto_increment";
```
Comment (RePod) [inline test.php]: Not switch-proof if somebody wants to convert, very bad.

File: _core/regist/regist.php (hunk 2 above)
Comment (Apogate) [inline _core/regist/regist.php]: From my looking around, `last_insert_id()` [only works on `AUTO INCREMENT`'ed fields](http://php.net/manual/en/function.mysql-insert-id.php#refsect1-function.mysql-insert-id-description), which for the post table to work, is not auto-incremented

**Comment by RePod** [comment]
Personally rejected, despite my comments.

Utilize MySQL's `UNION` to [combine the tables programmatically](//dev.mysql.com/doc/refman/5.7/en/union.html).
Worth keeping the `board` column though.

Obviously due to this, any functionality that relies on the result of the `UNION` should be its own file (or set of) and not injected throughout other files.

**Pull request closed without merging**